packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -431,14 +431,20 @@ export abstract class BaseSegment extends MergeNode implements ISegment {
         if (!this.properties) {
             this.properties = createMap<any>();
         }
-        return this.propertyManager.addProperties(
+        const result =  this.propertyManager.addProperties(
             this.properties,
             newProps,
             op,
             seq,
             collabWindow && collabWindow.collaborating,
             rollback,
         );
+
+        if (Object.entries(this.properties).length === 0) {
```

Comment:
take a look at matchProperties in properties.ts. i think that is probably the correct place to fix this.

Reply:
Is there a reason that changing matchProperties is a better fix than the change in addProperties? The bug description said that what I currently have is the preferred fix, but I can definitely change it if that makes more sense. (@scarlettjlee to weigh in as well)

Reply:
matchProperties would also be okay, IMO. But I do prefer to standardize the form of empty properties such that callers checking for them could just check for one thing.